src/any_guardrail/guardrails/llama_guard/llama_guard.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, ClassVar
+
+import torch
+from transformers import AutoModelForCausalLM, AutoProcessor, AutoTokenizer, Llama4ForConditionalGeneration
+
+from any_guardrail.base import GuardrailOutput
+from any_guardrail.guardrails.huggingface import HuggingFace
+
+
+class LlamaGuard(HuggingFace):
+    """Wrapper class for Llama Guard 3 & 4 implementations.
+
+    For more information about the implementations about either off topic model, please see the below model cards:
+
+    - [Meta Llama Guard 3 Docs](https://www.llama.com/docs/model-cards-and-prompt-formats/llama-guard-3/)
+    - [HuggingFace Llama Guard 3 Docs](https://huggingface.co/meta-llama/Llama-Guard-3-1B)
+    - [Meta Llama Guard 4 Docs](https://www.llama.com/docs/model-cards-and-prompt-formats/llama-guard-4/)
+    - [HuggingFace Llama Guard 4 Docs](https://huggingface.co/meta-llama/Llama-Guard-4-12B)
+    """
+
+    SUPPORTED_MODELS: ClassVar = [
+        "meta-llama/Llama-Guard-3-1B",
+        "meta-llama/Llama-Guard-3-8B",
+        "hf-internal-testing/tiny-random-LlamaForCausalLM",
+        "meta-llama/Llama-Guard-4-12B",
+    ]
+
+    def __init__(self, model_id: str | None = None) -> None:
+        """Llama guard model. Either Llama Guard 3 or 4 depending on the model id. Defaults to Llama Guard 3."""
+        self.model_id = model_id or self.SUPPORTED_MODELS[0]
+        if self.model_id in self.SUPPORTED_MODELS[-1]:
+
+            def _load_model(self) -> None:  # type: ignore[no-untyped-def]
+                self.tokenizer = AutoProcessor.from_pretrained(self.model_id)  # type: ignore[no-untyped-call]
+                self.model = Llama4ForConditionalGeneration.from_pretrained(self.model_id, dtype=torch.bfloat16)
+
+            self._load_model = _load_model.__get__(self)  # type: ignore[method-assign]
```

Comment:
This feels a little parkour.
Maybe we can just have a single `def self._load_model` defined at the class level and in this `if` we set something like `self._tokenizer_class` and `self._model_class`?

```python
def _load_model(self) -> None:
    self.tokenizer = self._tokenizer_class(self.model_id)
    self.model = self._model_class(self.model_id, dtype=torch.bfloat16)
```